### ISSUE TITLE
Overflow in multiply when formatting instruction

### ISSUE THREAD

**Issue by ghost** [iximeow/yaxpeax-x86#15]
(grr, the second time i hit enter and made an issue too early, sorry if you have email notifs on!)

```rust
fn main() {
    let decoder = yaxpeax_x86::amd64::InstDecoder::default();
    let mut s = String::new();
    let i = decoder.decode_slice(&[243, 15, 30, 15]).expect("unreachable");
    i.write_to(&mut s).expect("unreachable");
}
```

Gives a multiply with overflow with integer overflow checks on, gives a index out of bounds panic (thankfully not *unchecked* indexing) in release

**Comment by ghost** [comment]
The fuzzer I used to find this is just

```rust
#![no_main]
use libfuzzer_sys::fuzz_target;

fuzz_target!(|data: &[u8]| {
    let decoder = yaxpeax_x86::amd64::InstDecoder::default();
    let mut s = String::new();
    if let Ok(i) = decoder.decode_slice(data) {
        i.write_to(&mut s).expect("writing to never fail");
    }
});
```

**Comment by iximeow** [comment]
i see this yield a `attempt to subtract with overflow`, rather than multiply? the cause of this is a little deeper than just the formatting though: the `nop` instruction is reported as having a memory operand, and then never gets an explicit `mem_size`, so it keeps the default of 0. then in..
```rust
out.write_str(MEM_SIZE_STRINGS[instr.mem_size as usize - 1])
```
you can see why it would panic .. :D 

fixed in [`e7dec7b`](https://github.com/iximeow/yaxpeax-x86/commit/e7dec7baa9c6649d71e1b349d93dce6b0cd588bf).

and then i've added fuzz targets for both decoding and displaying instructions in [`26e019c`](https://github.com/iximeow/yaxpeax-x86/commit/26e019cc3788b6bac73969dc3d1753e883961339). i'm gonna let `cargo fuzz` go a while longer before cutting a new release, though, since i doubt this is the _only_ instruction i missed when adding the explicit `mem_size` field.
